TheApiAwakens/StarWarsDetailViewController.swift

```diff
@@ -0,0 +1,551 @@
+    @IBOutlet weak var smallestLabel: UILabel!
+    @IBOutlet weak var smallestLabelValue: UILabel!
+    @IBOutlet weak var largestLabelValue: UILabel!
+    @IBOutlet weak var usdButton: UIButton!
+    @IBOutlet weak var creditButton: UIButton!
+    @IBOutlet weak var englishButton: UIButton!
+    @IBOutlet weak var metricButton: UIButton!
+    @IBOutlet weak var exchangeLabel: UILabel!
+    @IBOutlet weak var exchangeTextField: UITextField!
+    @IBOutlet weak var pickerView: UIPickerView!
+
+
+    var didCharacterStarshipFinished = false
+    var totalValueToEnglish = 0.0
+    var totalValueToMetric = 0.0
+    var objectQuantity = 0
+    var pickerData = [String]()
+    var nextPageNumber = 1
+    var type = ResourceType.none
+    var swAPIClient: SWApiClient = SWApiClient()
+    var starships = [Starship]()
+    var vehicles = [Vehicle]()
+    var characters = [Character]()
+    var valueSelectedAllTypes: Measurable!
+    var valueSelectedTransportCraft: TransportCraft!
+    var isApiFirstCall = true
+    var characterSelected: Character!
+    var charactersVehicleAndStarship = ""
+    var starshipsIDs = [String]()
+    var vehiclesIDs = [String]()
+    var hasFinishedRequest = false
+
+    var planets = [Planet]() {
+        didSet {
+            setCharactersPlanetLabel()
+        }
+    }
+    var charactersVehicles = [Vehicle]() {
+        didSet {
+            setCharacterVehicleAndStarshipLabel()
+        }
+    }
+    var characterStarships = [Starship]() {
+        didSet {
+            setCharacterVehicleAndStarshipLabel()
+        }
+    }
+    var hasNextPage = false {
+        didSet {
+            if hasNextPage { nextPageNumber += 1 }
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        loadData()
+        pickerView.delegate = self
+        pickerView.dataSource = self
+        exchangeTextField.delegate = self
+        self.addDoneButtonOnKeyboard()
+    }
+
+    func loadData() {
+        switch type {
+            case .starship :
+                self.navigationItem.title = ResourceType.starship.rawValue
+                hideCharacterLabels()
+                fetchForStarship(with: nextPageNumber)
+            case .vehicle :
+                self.navigationItem.title = ResourceType.vehicle.rawValue
+                hideCharacterLabels()
+                fetchForVehicle(with: nextPageNumber)
+            case .character :
+                self.navigationItem.title = ResourceType.character.rawValue
+                fetchForCharacter(with: nextPageNumber)
+            default: break
+        }
+    }
+
+    func hideCharacterLabels() {
+        charactersVehicleStarshipLabel.isHidden = true
+        charactersVehicleStarshipValueLabel.isHidden = true
+    }
+
+    func fetchForStarship(with page: Int) {
+        swAPIClient.fetchForStarship(nextPage: page, completion: { [weak self] (result) in
+            switch result {
+                case .failure(let error):
```

Comment:
You are on the right track. You should indeed do the judgment you did.

But I think that if you're fetching the first page and that call fails, then your page will be empty. So you should show something to the user. 

If it's not the first page, then it doesn't matter.